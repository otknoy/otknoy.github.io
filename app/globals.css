body {
  background-color: #000020; /* Dark blue */
  color: #00f0f0; /* Cyan / Aqua */
  text-shadow: 0 0 5px #00f0f0, 0 0 10px #00f0f0, 0 0 15px #00f0f0; /* Glowing effect */
  font-family: 'Orbitron', sans-serif;
}

a {
  color: #00f0f0; /* Cyan / Aqua */
  text-decoration: none;
  transition: color 0.3s, text-shadow 0.3s; /* Added transition */
}

a:hover {
  color: #ffffff;
  text-shadow: 0 0 5px #ffffff, 0 0 10px #ffffff; /* Keep existing glow for general links */
}

.content-wrapper {
  max-width: 800px;
  margin: 2rem auto; /* Added more margin to top/bottom */
  padding: 2rem;
  border: 1px solid #00f0f0; /* Cyan border */
  box-shadow: 0 0 15px rgba(0, 240, 240, 0.3); /* Cyan glow effect */
  border-radius: 5px; /* Slightly rounded corners */
}

section {
  margin-bottom: 2rem; /* Add space between sections */
}

/* Add some basic styling for headings to fit the theme */
h1, h2, h3, h4, h5, h6 {
  color: #00f0f0; /* Cyan / Aqua */
  text-shadow: 0 0 3px #00f0f0; /* Subtle glow */
}

/* Header link styles */
header a {
  margin: 0 10px;
  padding: 5px 10px;
  border: 1px solid #00f0f0;
  border-radius: 3px;
  transition: background-color 0.3s, color 0.3s, box-shadow 0.3s;
}

header a:hover {
  background-color: #00f0f0;
  color: #000020; /* Dark blue, to contrast with cyan background */
  box-shadow: 0 0 10px #00f0f0;
  text-shadow: none; /* Remove text glow when background is glowing */
}

/* Skills list item styles */
section ul li { /* This might be too generic, ideally Skills component would have its own classes */
  border: 1px dashed rgba(0, 240, 240, 0.5); /* Dashed cyan border */
  padding: 0.5rem;
  margin-bottom: 0.5rem;
  border-radius: 3px;
  list-style-type: none; /* Remove default bullet points */
}
