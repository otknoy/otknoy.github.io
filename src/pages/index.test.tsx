import * as React from 'react'

import * as renderer from 'react-test-renderer'
import { jest, test, expect, describe } from '@jest/globals'

import Home from './index.page'

jest.useFakeTimers()
jest.setSystemTime(new Date('2025-01-01'))

describe('render <Home />', () => {
  test('snapshot', () => {
    const tree = renderer.create(<Home />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
