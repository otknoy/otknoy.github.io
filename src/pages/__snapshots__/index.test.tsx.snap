// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Home /> snapshot 1`] = `
<div>
  <header
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
  >
    <h1
      style={
        Object {
          "fontSize": "24px",
        }
      }
    >
      Naoya Otsuka
    </h1>
    <div>
      <ul
        style={
          Object {
            "listStyle": "none",
          }
        }
      >
        <li
          style={
            Object {
              "display": "inline-block",
              "padding": "4px",
            }
          }
        >
          <a
            href="https://github.com/otknoy"
          >
            <span
              style={
                Object {
                  "fontSize": "16px",
                }
              }
            >
              github
            </span>
          </a>
        </li>
        <li
          style={
            Object {
              "display": "inline-block",
              "padding": "4px",
            }
          }
        >
          <a
            href="https://twitter.com/otknoy"
          >
            <span
              style={
                Object {
                  "fontSize": "16px",
                }
              }
            >
              twitter
            </span>
          </a>
        </li>
        <li
          style={
            Object {
              "display": "inline-block",
              "padding": "4px",
            }
          }
        >
          <a
            href="https://otknoy.hatenablog.com"
          >
            <span
              style={
                Object {
                  "fontSize": "16px",
                }
              }
            >
              blog
            </span>
          </a>
        </li>
      </ul>
    </div>
  </header>
  <section>
    <h1
      style={
        Object {
          "fontSize": "22px",
        }
      }
    >
      Profile
    </h1>
    <div>
      <p>
        <span
          style={
            Object {
              "fontSize": "16px",
            }
          }
        >
          web engineer, architect
        </span>
      </p>
    </div>
  </section>
  <section>
    <h1
      style={
        Object {
          "fontSize": "22px",
        }
      }
    >
      Interest
    </h1>
    <ul>
      <li>
        <span
          style={
            Object {
              "fontSize": "16px",
            }
          }
        >
          Java, Kotlin, SpringBoot
        </span>
      </li>
      <li>
        <span
          style={
            Object {
              "fontSize": "16px",
            }
          }
        >
          Go, Docker, Kubernetes
        </span>
      </li>
      <li>
        <span
          style={
            Object {
              "fontSize": "16px",
            }
          }
        >
          Microservices, Test-driven Development
        </span>
      </li>
      <li>
        <span
          style={
            Object {
              "fontSize": "16px",
            }
          }
        >
          Information Retrieval
        </span>
      </li>
      <li>
        <span
          style={
            Object {
              "fontSize": "16px",
            }
          }
        >
          Human Computer Interaction
        </span>
      </li>
    </ul>
  </section>
  <section>
    <h1
      style={
        Object {
          "fontSize": "22px",
        }
      }
    >
      Background
    </h1>
    <section>
      <h1
        style={
          Object {
            "fontSize": "18px",
          }
        }
      >
        Master's Degree
      </h1>
      <span
        style={
          Object {
            "fontSize": "16px",
          }
        }
      >
        Graduate School of Informatics, Kansai University, 2013-2015
      </span>
    </section>
    <section>
      <h1
        style={
          Object {
            "fontSize": "18px",
          }
        }
      >
        Bachelor's Degree
      </h1>
      <span
        style={
          Object {
            "fontSize": "16px",
          }
        }
      >
        Faculty of Informatics, Kansai University, 2010-2013
      </span>
    </section>
  </section>
  <footer
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <h1
      style={
        Object {
          "margin": "0px",
          "padding": "4px",
        }
      }
    >
      <span
        style={
          Object {
            "fontSize": "16px",
          }
        }
      >
        (C) 1988-
        2025
         Naoya Otsuka
      </span>
    </h1>
  </footer>
  <style
    jsx={true}
  >
    
      div {
        max-width: 410px;
        margin: 0 auto;
        padding: 0 8px;
      }
      
  </style>
</div>
`;
